lms/views/admin.py

```diff
@@ -34,52 +35,83 @@ def index(self):
 
     @view_config(
         route_name="admin.instances",
+        request_method="GET",
         renderer="lms:templates/admin/instances.html.jinja2",
     )
     def instances(self):  # pylint: disable=no-self-use
         return {}
 
     @view_config(
-        route_name="admin.instances",
+        route_name="admin.instances.search",
         request_method="POST",
         require_csrf=True,
+        renderer="lms:templates/admin/instances.results.html.jinja2",
     )
-    def find_instance(self):
-        try:
-            consumer_key = self.request.params["query"]
-        except KeyError as err:
-            raise HTTPBadRequest() from err
-
-        try:
-            ai = self.application_instance_service.get_by_consumer_key(consumer_key)
-        except ApplicationInstanceNotFound:
+    def search(self):
+        if not any(
+            (
+                self.request.params.get(param)
+                for param in [
+                    "consumer_key",
+                    "issuer",
+                    "client_id",
+                    "deployment_id",
+                    "tool_consumer_instance_guid",
+                ]
+            )
+        ):
             self.request.session.flash(
-                f'No application instance found for {self.request.params["query"]}',
-                "errors",
+                "Need to pass at least one search criteria", "errors"
             )
             return HTTPFound(location=self.request.route_url("admin.instances"))
 
-        return HTTPFound(
-            location=self.request.route_url(
-                "admin.instance", consumer_key=ai.consumer_key
+        instances = self.application_instance_service.search(
+            consumer_key=self.request.params.get("consumer_key"),
+            issuer=self.request.params.get("issuer"),
+            client_id=self.request.params.get("client_id"),
+            deployment_id=self.request.params.get("deployment_id"),
+            tool_consumer_instance_guid=self.request.params.get(
+                "tool_consumer_instance_guid"
             ),
         )
 
+        if not instances:
+            self.request.session.flash("No instances found", "errors")
+            return HTTPFound(location=self.request.route_url("admin.instances"))
+
+        if len(instances) == 1:
```

Comment:
I think I prefer this behavior, I'm searching for something that identifies only one AI I rather be taken directly to it.

Some fields do indeed always identify an AI (consumer_key) while other might do "often" (the LTI1.3 ones, deployment_id).